docs/about/index.rst

```diff
@@ -17,13 +40,156 @@ Code of conduct
+beyond binary classification.
+
+The project was greatly expanded in the second half of 2019 thanks to the
+involvement of many contributors from Azure ML and Microsoft Research.
+At that time, the project started to have regular releases.
+
 .. _roadmap:
 
 Project roadmap
 ---------------
 
+As an open-source project, Fairlearn strives to incorporate the best of
+research and practice.  AI is a rapidly evolving field, and fairness in AI is
+all the more so. We therefore encourage researchers, practitioners, and other
+stakeholders to contribute fairness metrics, unfairness mitigation algorithms,
+and visualization capabilities to Fairlearn as we experiment, learn, and
+evolve the project together.
+
+There are many areas for future enhancement and growth. For example, Fairlearn
+currently supports only *group fairness*&mdash;i.e., fairness with respect to
+groups of people, such as those defined in terms of race, sex, age, or
+disability status&mdash;and not other conceptualizations of fairness, such as
+*individual fairness* or *counterfactual fairness.* Fairlearn also currently
+includes only a limited set of unfairness mitigation algorithms, although we
+note that these algorithms are not restricted to classification tasks. Besides
+adding fairness metrics, conceptualizations of fairness, and unfairness
+mitigation algorithms, we also hope that Fairlearn will expand to cover more
+complex machine learning tasks in areas like counterfactual reasoning,
+computer vision, and natural language processing.
+We also anticipate integrating Fairlearn with interpretability tools, such as
+`InterpretML <https://github.com/interpretml>`_.
+
+Ultimately, we hope that Fairlearn will become more than a software
+tool&mdash;a vibrant community and resource center that provides not only
+code, but also resources like domain-specific guides for when and when not to
+use different fairness metrics and unfairness mitigation algorithms.
+
+Short-term roadmap
+^^^^^^^^^^^^^^^^^^
+
+*Last update: June 19, 2020*
+
+Here we list some specific areas we would like to work on in the next few
+weeks and months. If you want to get involved, please
+:ref:`reach out <communication>`.  The inclusion of tasks in the list below
+does not mean that we are committing to deliver them by a specific date.
+It merely represents a current list of tasks that members of the community
+work on. If you are interested to work on a topic that is not covered, please
+reach out.
+
+Tasks
+~~~~~
+
+#. *Improve the `fairlearn.metrics` API*
+   More information in the
+   `fairlearn-proposals <https://github.com/fairlearn/fairlearn-proposals/blob/master/api/METRICS.md>`_
+   repository.
+
+   **How you can help:** critique the proposal, suggest new metrics, work on
+   implementing new metrics.
+
+#. *Improve the project documentation*
+   The Fairlearn website is still very new and raw with lots of gaps.
+   `This proposal <https://github.com/fairlearn/fairlearn-proposals/pull/8>`_
```

Comment:
I agree! Proposals are just for that: proposing. Beyond that the PRs should get checked in and the work happens in this repo. Any update would require a PR, so that's a huge barrier to updating things from "TODO" to "in progress" or something like that.

Issues don't lend themselves to this either, though, because it's a thread (even though you can change the content of certain messages). I guess we could use them that way, but it might confuse people, too. 

The concept that is meant to track these sort of things in GitHub is the milestone. We just haven't used it properly so far. I'll try and clean that up later today and perhaps that helps? Besides having that organized I also think we need to be more disciplined in marking things as "help wanted" (there's a tag for that) so that it's clear what people can pick up, and marking items as "taken" (by assigning it to ourselves), so that people know what's already currently being worked on. I'll spend a bit of time cleaning this up today.

Reply:
UPDATE: milestones do make everything a bit hard to read since they don't properly support markdown. Lists (and task lists that you use above) don't render there, but they do in PR comments and issues. So here's a (hopefully) better proposal: 

For every workstream (i.e., more than 1 issue) we create an "high-level" issue and tag it as "epic" (or something similar). That epic issue then contains the task list (with links to all sub-issues) and outlines if there are dependencies. The tag "help wanted" on each issue indicates whether the owner of the epic would like others to help. That should clear confusion on where it's easy to contribute.

[If we go with this I'd like to get rid of all milestones we have, except perhaps for the one that tracks the next release. I've read in a few places now that they're meant for sprints as opposed to epics, so that explains why we can assign issues only to a single sprint.]

FYI @MiroDudik @riedgar-ms

Reply:
I am worried about the maintenance overhead (similarly to @vingu below), but I like the suggestion to use the project board (also suggested by @mbrouns ) or even multiple project boards. We need something like that anyway to organize current issues and PRs--see, e.g., https://github.com/scikit-learn/scikit-learn/projects . @adrinjalali -- what has been your experience with these?

If there is an obvious set of items / tasks that comes up within an issue or a proposal, then it's probably not too much overhead to create more granular issues and tag them as "high-priority" (if that's the case) and/or "help wanted"--and put them in the same project as the proposal. We probably shouldn't mark "TODOs" in the proposal and instead just rely on issues (that was my mistake...). If there ends up being a single issue attached to implementation of a proposal, I think that Kevin's suggestion might be easily done by presenting the most up-to-date TODO list at the bottom of the issue--this could be the same comment that's re-edited, or simply the latest comment.

That said--I don't want to force the author of an issue / proposal to split it into smaller tasks if that presents an overhead. I'd rather postpone splitting until there's somebody who wants to help and let them figure it out with whoever submitted the issue (or maybe somebody like the "project owner" if we end up having those).

I feel similar about having a "high-level" issue tagged as "epic" in a project--I'm not opposed, but also don't want to force it, for fear of creating additional overhead.

My current preference is to experiment with project boards and labels (I'll start a new issue). This will help us manage the workflow and ever-so-slightly help with the "discoverability" of the more granular tasks in support of the already planned work. I think it will render some of the material in this "Short-term roadmap" redundant, and instead we can then rewrite the content here to complement the project boards rather than duplicate them.

I like the approach suggested by @kevinrobinson [above](https://github.com/fairlearn/fairlearn/pull/500#discussion_r443040820) -- if we describe the real-world problems we're trying to address (both short-term and long-term), the contributors will be able to come up with issues and/or projects themselves and we will have an easier time identifying the "high-priority" issues. I'm not yet sure if we should use these "problems" as "projects" in our project boards... I think we'll need to experiment a bit.

Reply:
[moving the discussion of the project boards structure to #504 ]